front/src/reducers/index.ts

```diff
@@ -65,30 +73,38 @@ const saveMainFilter = createFilter('main', mainWhiteList);
 const saveSimulationFilter = createFilter('osrdsimulation', simulationWhiteList);
 
 // Useful to only blacklist a sub-propertie of osrdconf
-const osrdconfPersistConfig: PersistConfig<OsrdMultiConfState> = {
-  key: 'osrdconf',
+const buildOsrdConfPersistConfig = <T extends OsrdConfState>(
+  slice: operationalStudiesConfSliceType | stdcmConfSliceType
+): PersistConfig<T> => ({
+  key: slice.name,
   storage,
-  blacklist: ['featureInfoClick', 'switchTypes'],
-};
+  blacklist: ['featureInfoClick'],
+});
 
 export const persistConfig = {
   key: 'root',
   storage,
   transforms: [compressor, saveMapFilter, saveUserFilter, saveMainFilter, saveSimulationFilter],
-  blacklist: ['osrdconf'],
-  whitelist: ['user', 'map', 'main', 'simulation'],
+  blacklist: [stdcmConfSlice.name, operationalStudiesConfSlice.name],
+  whitelist: ['user', 'map', 'main', 'simulation', 'mapViewer'],
 };
 
-type AllActions = EditorActions | Action;
+type AllActions = Action;
+
+export type OsrdSlicesType =
+  | operationalStudiesConfSliceType
+  | stdcmConfSliceType
+  | editorSliceType
+  | mapViewerSliceType;
 
 export interface RootState {
   user: UserState;
   map: MapState;
-  [editorSlice.name]: EditorState;
+  mapViewer: MapViewerStateType;
```

Comment:
Can you harmonize here (same for user, map and main):
```suggestion
  [mapViewerSlice.name]: MapViewerStateType;
```

Reply:
Done